# **1. Perkenalan Dataset**


Tahap pertama, Anda harus mencari dan menggunakan dataset dengan ketentuan sebagai berikut:

1. **Sumber Dataset**:  
   Dataset dapat diperoleh dari berbagai sumber, seperti public repositories (*Kaggle*, *UCI ML Repository*, *Open Data*) atau data primer yang Anda kumpulkan sendiri.


# **2. Import Library**

Pada tahap ini, Anda perlu mengimpor beberapa pustaka (library) Python yang dibutuhkan untuk analisis data dan pembangunan model machine learning atau deep learning.

In [3]:
%pip install pandas numpy matplotlib seaborn scikit-learn ipykernel -q

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 26.0.1 -> 26.1.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [ ]:
# Pustaka utama untuk manipulasi dan analisis data tabel
import pandas as pd
import numpy as np

# Pustaka untuk visualisasi data
import matplotlib.pyplot as plt
import seaborn as sns

# Pustaka untuk preprocessing 
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler, OneHotEncoder

# Mengatur agar visualisasi langsung muncul di notebook dan terlihat rapi
%matplotlib inline
sns.set_theme(style="whitegrid")

# **3. Memuat Dataset**

Pada tahap ini, Anda perlu memuat dataset ke dalam notebook. Jika dataset dalam format CSV, Anda bisa menggunakan pustaka pandas untuk membacanya. Pastikan untuk mengecek beberapa baris awal dataset untuk memahami strukturnya dan memastikan data telah dimuat dengan benar.

Jika dataset berada di Google Drive, pastikan Anda menghubungkan Google Drive ke Colab terlebih dahulu. Setelah dataset berhasil dimuat, langkah berikutnya adalah memeriksa kesesuaian data dan siap untuk dianalisis lebih lanjut.

Jika dataset berupa unstructured data, silakan sesuaikan dengan format seperti kelas Machine Learning Pengembangan atau Machine Learning Terapan

In [3]:
import os

# 1. Tentukan path/lokasi
file_path = os.path.join('..', 'Titanic dataset_raw.csv')

# 2. Memuat file CSV menggunakan pandas
try:
    df = pd.read_csv(file_path)
    print("✅ Dataset berhasil dimuat dengan sukses!")
except FileNotFoundError:
    print("❌ File tidak ditemukan! Periksa kembali apakah nama file dan foldernya sudah benar.")

# 3. Menampilkan 5 baris pertama
df.head()

✅ Dataset berhasil dimuat dengan sukses!


,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,892,0,3,"Kelly, Mr. James",male,34.5,0,0,330911,7.8292,NaN,Q
1,893,1,3,"Wilkes, Mrs. James (Ellen Needs)",female,47.0,1,0,363272,7.0000,NaN,S
2,894,0,2,"Myles, Mr. Thomas Francis",male,62.0,0,0,240276,9.6875,NaN,Q
3,895,0,3,"Wirz, Mr. Albert",male,27.0,0,0,315154,8.6625,NaN,S
4,896,1,3,"Hirvonen, Mrs. Alexander (Helga E Lindqvist)",female,22.0,1,1,3101298,12.2875,NaN,S


# **4. Exploratory Data Analysis (EDA)**

Pada tahap ini, Anda akan melakukan **Exploratory Data Analysis (EDA)** untuk memahami karakteristik dataset.

Tujuan dari EDA adalah untuk memperoleh wawasan awal yang mendalam mengenai data dan menentukan langkah selanjutnya dalam analisis atau pemodelan.

In [14]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    str    
 4   Sex          418 non-null    str    
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    str    
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     str    
 11  Embarked     418 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 39.3 KB


# **5. Data Preprocessing**

Pada tahap ini, data preprocessing adalah langkah penting untuk memastikan kualitas data sebelum digunakan dalam model machine learning.

Jika Anda menggunakan data teks, data mentah sering kali mengandung nilai kosong, duplikasi, atau rentang nilai yang tidak konsisten, yang dapat memengaruhi kinerja model. Oleh karena itu, proses ini bertujuan untuk membersihkan dan mempersiapkan data agar analisis berjalan optimal.

Berikut adalah tahapan-tahapan yang bisa dilakukan, tetapi **tidak terbatas** pada:
1. Menghapus atau Menangani Data Kosong (Missing Values)
2. Menghapus Data Duplikat
3. Normalisasi atau Standarisasi Fitur
4. Deteksi dan Penanganan Outlier
5. Encoding Data Kategorikal
6. Binning (Pengelompokan Data)

Cukup sesuaikan dengan karakteristik data yang kamu gunakan yah. Khususnya ketika kami menggunakan data tidak terstruktur.

In [15]:
print("Jumlah data kosong:")
print(df.isnull().sum())
print(f"\nJumlah data duplikat: {df.duplicated().sum()}")
print("-" * 50)



Jumlah data kosong:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

Jumlah data duplikat: 0
--------------------------------------------------


#### Analisis & Penanganan Missing Values:
1. Kolom Cabin: Dihapus (drop) dari dataset karena persentase data yang hilang mencapai >78% (327 dari 418 baris). Mengisi data dengan tebakan pada skala sekecil ini dapat memanipulasi keaslian informasi.
2. Kolom Age & Fare: Mengandung data kosong dalam jumlah minor, sehingga diimputasi menggunakan nilai Median agar tidak sensitif terhadap outlier.

In [16]:
df = df.drop(columns=['Cabin'], errors='ignore') 
df['Age'] = df['Age'].fillna(df['Age'].median())  
df['Fare'] = df['Fare'].fillna(df['Fare'].median()) 



In [17]:
print("Jumlah data kosong:")
print(df.isnull().sum())
print(f"\nJumlah data duplikat: {df.duplicated().sum()}")

Jumlah data kosong:
PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

Jumlah data duplikat: 0


In [19]:
# Kolom Nama, ID, dan Nomor Tiket tidak mempengaruhi keselamatan
kolom_tidak_butuh = ['PassengerId', 'Name', 'Ticket']
df_selected = df.drop(columns=kolom_tidak_butuh, errors='ignore')

# Mengubah kolom 'Sex' dan 'Embarked' menjadi angka biner (0 dan 1) agar bisa dibaca model ML
df_ready = pd.get_dummies(df_selected, columns=['Sex', 'Embarked'], drop_first=True)


print("✅ Preprocessing Selesai!")
print(f"Ukuran dataset saat ini: {df_ready.shape[0]} baris dan {df_ready.shape[1]} kolom.\n")
df_ready.head()

✅ Preprocessing Selesai!
Ukuran dataset saat ini: 418 baris dan 9 kolom.



,Survived,Pclass,Age,SibSp,Parch,Fare,Sex_male,Embarked_Q,Embarked_S
0,0,3,34.5,0,0,7.8292,True,True,False
1,1,3,47.0,1,0,7.0000,False,False,True
2,0,2,62.0,0,0,9.6875,True,True,False
3,0,3,27.0,0,0,8.6625,True,False,True
4,1,3,22.0,1,1,12.2875,False,False,True
